### ISSUE TITLE
Series got grouped into wrong Season

### ISSUE THREAD

**Issue by t0815** [ShokoAnime/ShokoServer#939]
## VERSION INFORMATION

Server Version: 4.1.2.0

Desktop Version: 4.1.2.0

If you've compiled your own version, please use the last commit you compiled.

## LOG FILE

n/a

## DESCRIPTION

Series  "Kakkou no Iinazuke ([a16188](https://anidb.net/a16188))" got grouped into wrong Season now, (after latest Update)  all of a sudden. It was in the correct season "Spring 2022" before, but after update it got moved into Season "Summer 2022".

![shoko_wrong_season](https://user-images.githubusercontent.com/10379828/177042808-fcedb589-f7e9-4cda-8b45-f1c71441bd98.jpg)

Its not present in "Spring 2022" anymore.

![shoko_wrong_season_2](https://user-images.githubusercontent.com/10379828/177042999-ba10b61f-a914-45b7-8490-0b90f0480c38.jpg)

## STEPS TO REPRODUCE

1. Have Series imported
2. check apprpriate season filters

**Comment by Cazzar** [comment]
Looking at the history of this anime, it was modified a few times with the data updated, last about 3 months ago, have you seen if "Update AniDB info" will do anything?

**Comment by t0815** [comment]
Yes did try that too, no change.

**Comment by t0815** [comment]
Show seems to be 2-Cour show tho, so its ok to be on Summer season too.

**Comment by da3dsoul** [comment]
A bit late to the party. I didn't see this. So, firstly, the season label may not match AniDB. They don't provide that info in the API. The math behind whether a series should be considered is as such. If you don't want to/can't read code, just ignore it and read the comments.

This was put together by my reasoning, and it's likely to be correct by my logic, even if other sources say otherwise. After all, a "season" is only helpful to viewers as an indication of when something airs.
If you note a flaw in the logic, tell me here. If it's another issue, like the group filter simply not updating when it should, also tell me.

Based on the 2 series listed and the info available at the time, I think it's fair to call them as they are listed. You are correct that series without an End Date listed will temporarily show up in incorrect seasons until they are updated. That can't be fixed, as the data is missing. I would check them now to see if after these few months have passed, the data has been corrected, and they have rearranged themselves into the right places.

```cs
        public static bool IsInSeason(this AniDB_Anime anime, AnimeSeason season, int year)
        {
            if (anime.AirDate == null) return false;
            // If it isn't a normal series, then it won't adhere to standard airing norms
            if (anime.AnimeType != (int) AnimeType.TVSeries && anime.AnimeType != (int) AnimeType.Web) return false;
            DateTime seasonStartBegin;
            DateTime seasonStartEnd;
            switch (season)
            {
                case AnimeSeason.Winter:
                    // January +- 0.5
                    seasonStartBegin = new DateTime(year - 1, 12, 16);
                    seasonStartEnd = new DateTime(year, 1, 15);
                    break;
                case AnimeSeason.Spring:
                    // April +- 0.5
                    seasonStartBegin = new DateTime(year, 3, 16);
                    seasonStartEnd = new DateTime(year, 4, 15);
                    break;
                case AnimeSeason.Summer:
                    // July +- 0.5
                    seasonStartBegin = new DateTime(year, 6, 15);
                    seasonStartEnd = new DateTime(year, 7, 15);
                    break;
                case AnimeSeason.Fall:
                    // October +- 0.5
                    seasonStartBegin = new DateTime(year, 9, 15);
                    seasonStartEnd = new DateTime(year, 10, 15);
                    break;
                default:
                    return false;
            }
            // Don't even count seasons that haven't happened yet
            if (seasonStartBegin > DateTime.Today) return false;

            // If it starts in a season, then it is definitely going to be in it
            if (anime.AirDate.Value >= seasonStartBegin && anime.AirDate.Value <= seasonStartEnd) return true;
            // If it aired before the season, but hasn't finished by the season, count it.
            if (anime.AirDate.Value < seasonStartBegin)
            {
                // null EndDate means it's still airing now
                if (anime.EndDate == null) return true;
                // A season can run long, so don't count it unless it continues well into the season
                switch (season)
                {
                    case AnimeSeason.Winter:
                        // January + 1
                        seasonStartBegin = new DateTime(year, 2, 1);
                        break;
                    case AnimeSeason.Spring:
                        // April + 1
                        seasonStartBegin = new DateTime(year, 5, 1);
                        break;
                    case AnimeSeason.Summer:
                        // July + 1
                        seasonStartBegin = new DateTime(year, 8, 1);
                        break;
                    case AnimeSeason.Fall:
                        // October + 1
                        seasonStartBegin = new DateTime(year, 11, 1);
                        break;
                    default:
                        return false;
                }
                if (anime.EndDate.Value > seasonStartBegin) return true;
            }
            return false;
        }
```

**Comment by ElementalCrisis** [comment]
@revam are you still looking into this?

**Comment by revam** [comment]
waiting on anidb to expose their season data to fix it for good (a bug report has already been filed), but until then it will remain flawed(?) as it is auto-generated based off dates.
